test/e2e/logging_test.go

```diff
@@ -68,9 +69,16 @@ func TestRequestLogs(t *testing.T) {
 		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
 	}
 
-	_, err = sendRequest(t, clients, test.ServingFlags.ResolvableDomain, resources.Route.Status.URL.URL())
+	_, err = pkgTest.WaitForEndpointState(
```

Comment:
I still don't know why we need this.

Reply:
Because downstream projects might use external domains that actually need DNS setup for example, or external loadbalancers. It's a bandaid I brought up multiple times maybe I should bring it up to Networking WG again. It needs to be well understood and harder to hold wrong.

Reply:
I think you were thinking of keeping the bandaid behind a flag?

Reply:
Could you add a comment for it?

Reply:
There's an open PR to add the flag! https://github.com/knative/serving/pull/7390